rfhub2/cli/rfhub_importer.py

```diff
@@ -0,0 +1,216 @@
+from pathlib import Path
+import re
+from robot.errors import DataError
+from robot.libdocpkg import LibraryDocumentation
+import robot.libraries
+from typing import Dict, List, Set, Tuple
+
+from .api_client import Client
+from rfhub2.model import Collection, Keyword
+
+
+RESOURCE_PATTERNS = {'.robot', '.txt', '.tsv', '.resource'}
+ALL_PATTERNS = (RESOURCE_PATTERNS | {'.xml', '.py'})
+EXCLUDED_LIBRARIES = {'remote.py', 'reserved.py', 'dialogs.py', 'dialogs_jy.py',
+                      'dialogs_py.py', 'dialogs_ipy.py', 'setup.py'}
+INIT_FILES = {'__init__.txt', '__init__.robot', '__init__.html', '__init__.tsv'}
+
+
+class RfhubImporter(object):
+
+    def __init__(self, paths: Tuple[Path, ...], no_installed_keywords: bool, client: Client) -> None:
+        self.paths = paths
+        self.no_installed_keywords = no_installed_keywords
+        self.client = client
+
+    def delete_collections(self) -> Set[int]:
+        """
+        Deletes all existing collections.
+        """
+        self.client.check_communication_with_app()
+        collections = self.client.get_collections()
+        collections_id = {collection['id'] for collection in collections}
+        for id in collections_id:
+            self.client.delete_collection(id)
+        return collections_id
+
+    def import_libraries(self) -> Tuple[int, int]:
+        """
+        Import libraries to application from paths specified when invoking client.
+        :return: Number of libraries loaded
+        """
+        libraries_paths = self.get_libraries_paths()
+        collections = self.create_collections(libraries_paths)
+        loaded_collections = self.add_collections(collections)
+        return len(loaded_collections), sum(v for dict in loaded_collections for v in dict.values())
```

Comment:
you shadow `dict` class here. it is only local scope and it does not affect anything, but still it is rather bad convention.

Reply:
Fixed, but still:
![shame on me](https://media.makeameme.org/created/shame-on-me.jpg)